explorer/ast/expression.h

```diff
@@ -630,6 +630,12 @@ class OperatorExpression : public RewritableMixin<Expression> {
     return arguments_;
   }
 
+  // Can only be called by type-checking, if a conversion was required.
+  void set_argument(int index, Nonnull<Expression*> arg) {
```

Comment:
`arguments()` already exposes a MutableArrayRef, isn't this duplicative? i.e., `set_argument(i, arg)` -> `arguments()[i] = arg`